### ISSUE TITLE
[0.10.2] Redirect to leader not using proper header location in HTTP Headers?

### ISSUE THREAD

**Issue by smalenfant** [influxdata/influxdb#5928]
I've got a few problems with a cluster. Node 2 doesn't says it's part of the servers (show servers) but it doesn't share data with node 1 and 3. Not sure how to debug clustering, but I can't seem to resolve the problem, might have to restart install from scratch.

But here's what I'm seeing... If issue a curl command, I'm being redirect properly to the leader... 

```
[root@influx2 influxdb]# curl 'http://influx1:8091/lease?name=continuous_querier&nodeid=0'
<a href="http://influx2:8091/lease">Temporary Redirect</a>.
```

When I look at the logs on `influx2`, here is what I see 

```
[meta] 2016/03/07 18:30:19 10.22.250.31 - - [07/Mar/2016:18:30:19 +0000] GET /lease HTTP/1.1 400 44 http://influx1:8091/lease?name=continuous_querier&nodeid=0
```

Not sure why the redirect is not followed with the right host header information... Not sure if that's a problem with my configuration or not. But production works fine and data are being replicated, but the logs shows the same problem.

**Comment by smalenfant** [comment]
This seems to be related to #5592. How is this affecting the system? My leader has to be up for CQ to work?

**Comment by jonseymour** [comment]
Can you re-run the command against with curl's -v option?

**Comment by jonseymour** [comment]
I think the log entry you are seeing on influx2 is correct at least in respect of the host name in the full URL. The log entry is showing you the referrer URL which was http://influx1:8091/lease?name=continuous_querier&nodeid=0

**Comment by smalenfant** [comment]
You are right, it is the referer... Seems like the problem is that the new location doesn't include the proper query string... This should be fixed by #5592 which is exactly the same problem. Can this fix be merged in 0.10.x ?

```
[meta] 2016/03/08 06:01:03 172.16.18.4 - - [08/Mar/2016:06:01:03 +0000] GET /lease HTTP/1.1 400 44 http://influx1:8091/lease?name=continuous_querier&nodeid=1 Go 1.1 package http 23c821ff-e4f3-11e5-9c86-000000000000 1.792762ms
```

See the following...

```
# Failed by following the redirect
[root@influx2~]$ curl  'http://influx1.rd.at.cox.net:8091/lease?name=continuous_querier&nodeid=1' -L
lease name required

# Works, manually added the query string.
[root@influx2 ~]$ curl  'http://influx2:8091/lease?name=continuous_querier&nodeid=1'
{"name":"continuous_querier","expiration":"2016-03-08T06:03:44.464572241Z","owner":1}[root@influx2~]$
```

**Comment by jonseymour** [comment]
I've submitted PR #5933  to backport #5594 onto the 0.10.0 branch. It's not up to me to decide whether there will be a v0.10.3 release or whether this fix should be included in it but the PR and/or branch is available for use by anyone who needs it

**Comment by smalenfant** [comment]
Thanks. Let me try it out when I can. By the way, the problem kind of goes away once you restart each node in the cluster (all down, then bring up the leader first). Nut seems like something happens over time (or if a node restart).

**Comment by smalenfant** [comment]
Thanks, problem fixed in 0.10.3
